### PR TITLE
Add osu!direct download support

### LINKED ISSUE #1155: Add osu!direct download support

### PR DESCRIPTION
**Pull request by peppy**
Resolves #1155.

### CODE REVIEW

File: osu.Game/Overlays/Direct/DirectPanel.cs [hunk 1]
```diff
@@ -87,5 +260,30 @@ public Statistic(FontAwesome icon, int value = 0)
                 Value = value;
             }
         }
+
+        private class TransformEdgeEffectRadius : Transform<float, Container>
```
Comment (smoogipooo) [inline osu.Game/Overlays/Direct/DirectPanel.cs]: Move this to CompositeDrawable and add to TransformableExtensions + TransformSequence plz.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Overlays/Direct/DirectPanel.cs (hunk 1 above)
Comment (peppy) [inline osu.Game/Overlays/Direct/DirectPanel.cs]: are we seriously going to do this for every property of every property?

Comment (smoogipooo) [inline osu.Game/Overlays/Direct/DirectPanel.cs]: But this may be useful in other cases.

**Review by smoogipoo — APPROVED** [review]

File: osu.Game/Overlays/Direct/DirectPanel.cs (hunk 1 above)
Comment (Tom94) [inline osu.Game/Overlays/Direct/DirectPanel.cs]: We should have transform helpers for pretty much every property the framework exposes at least, imho.

Comment (peppy) [inline osu.Game/Overlays/Direct/DirectPanel.cs]: is planned for tomorrow

**Pull request merged**